Unpin python

znichollscr opened the pull request:
Unless there is a very good reason, all the advice I have read is to keep your pins as loose as possible. This article provides a very deep dive into it: https://iscinumpy.dev/post/bound-version-constraints/

SebastienGardoll commented:
Hi,

I agree, but, if I remember well, Laurent experienced an issue with its Linux environment and a version of Python (3.13 ?).
Laurent: do you still have this problem?

znichollscr commented:
> I agree, but, if I remember well, Laurent experienced an issue with its Linux environment and a version of Python (3.13 ?).
> Laurent: do you still have this problem?

Even if that's true, I would recommend letting Laurent (or any other user) make the choice of Python themselves, rather than pinning it here and effectively deciding for the user

SebastienGardoll commented:
Sure, but how set the Python version when we `rye sync`? I have to google that.

SebastienGardoll commented:
Found! `rye pin [PYTHON_VERSION]`. But:

- I think we should migrate to uv package manager. It also provides the `--python` option. @ltroussellier: if I remember, you choose rye because uv does not support script commands. But do you still use them (e.g., ev, evc, etc.)?
- ESGVOC use match/case statements which have been introduced by Python version 3.10. I am OK to remove the upper bound of the Python version in pyproject.toml, but I prefered to set the lower bound to 3.12.

SebastienGardoll commented:
@ltroussellier: it is possible to write a Makefile or use Hatch in order to execute your e* commands. I can write the Makefile for you.

znichollscr commented:
> I think we should migrate to uv package manager

If you're going to do this, this repo uses uv for everything and might help with examples for CI, Makefile etc. https://github.com/openscm/pandas-openscm. Underneath it, it uses this template (https://gitlab.com/znicholls/copier-core-python-repository) which will just set everything up for you and might save you some typing...

I'm also happy to just answer questions, I use uv for everything.

> uv does not support script commands

I'm puzzled by this, what exactly do you mean? uv has been developing very fast, so it's worth checking if they do support this now..

> Python version 3.10. I am OK to remove the upper bound of the Python version in pyproject.toml, but I prefered to set the lower bound to 3.12.

Can I just check: why would you pin the lower bound to 3.12 if the feature you need was introduced in Python 3.10 (maybe I am missing something)?

SebastienGardoll commented:
Hi @znichollscr,

- I have migrated the project to UV in the uv branch. I am waiting @ltroussellier for its validation. Thank your help.
- "UV doesn't support script commands": I mean script shortcut implemented with the pyproject section "[tool.rye.scripts]"
- I didn't find anything about UV and scripts last friday. But it is fine: makefile is great.
- You didn't miss anything about the Python version. It's just we developed ESGVOC with Python 3.12 and I don't want to implement tests and check if ESGVOC works with the versions 3.10.x and 3.11.x . Maybe I am wrong, and one can say that Python system is frequently < 3.12, and we would use the Python system, but ESGVOC needs libraries that are not installed in OS by default. I prefer to stick with the version of Python that I use for the development.

znichollscr commented:
> I didn't find anything about UV and scripts last friday. But it is fine: makefile is great

This issue (https://github.com/astral-sh/uv/issues/5903) was all I could find. Seems like there are some solutions/workarounds, but none as straightforward as what rye had.

> You didn't miss anything about the Python version...

Ok. I would be tempted to still not pin in pyproject.toml, because then it means you cannot install it for Python 3.10 (even if you're willing to take the risk that things don't work). But, it also doesn't really matter for me that much (I will probably only ever use esgvoc standalone so I can always use Python 3.12 if I need to).

SebastienGardoll commented:
Hi @znichollscr,

Laurent has validated my uv branch. I have just merged the uv branch into our integration branch. UV is now the package manager. As I have changed the Python version, this PR is no longer relevent.

znichollscr commented:
All good, thanks

On Wed, Apr 23, 2025, 11:10 Sébastien Gardoll ***@***.***>
wrote:

> Hi @znichollscr <https://github.com/znichollscr>,
>
> Laurent has validated my uv branch. I have just merged the uv branch into
> our integration branch. UV is now the package manager. As I have changed
> the Python version, this PR is no longer relevent.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ESGF/esgf-vocab/pull/17#issuecomment-2823601578>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A3KEXHZUBR3BQ6DK3DVNHRD225KH5AVCNFSM6AAAAAB2623LPSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQMRTGYYDCNJXHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> *SebastienGardoll* left a comment (ESGF/esgf-vocab#17)
> <https://github.com/ESGF/esgf-vocab/pull/17#issuecomment-2823601578>
>
> Hi @znichollscr <https://github.com/znichollscr>,
>
> Laurent has validated my uv branch. I have just merged the uv branch into
> our integration branch. UV is now the package manager. As I have changed
> the Python version, this PR is no longer relevent.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ESGF/esgf-vocab/pull/17#issuecomment-2823601578>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A3KEXHZUBR3BQ6DK3DVNHRD225KH5AVCNFSM6AAAAAB2623LPSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQMRTGYYDCNJXHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
